### PR TITLE
Qgis add missing attributes

### PR DESCRIPTION
**Pull request by jirik**
- [x] Tests
- ~~Layman Test Client~~
- ~~Changlelog~~
- [x] Documentation

### CODE REVIEW

File: doc/rest.md [hunk 1]
```diff
@@ -226,12 +227,13 @@ Body parameters:
       - file names, i.e. array of strings
    - if file names are provided, files must be uploaded subsequently using [POST Layer Chunk](#post-layer-chunk)
    - if published file has empty bounding box (i.e. no features), its bounding box on WMS/WFS endpoint is set to the whole World
+   - if QML style is used (either directly within this request, ir indirectly from previous state on server), it must list all attributes contained in given data file
```
Comment (index-git) [inline doc/rest.md:230]: => 'if QML style is used (either directly within this request, **or** indirectly from previous state on server), it must list all attributes contained in given data file'

File: src/layman/layer/qgis/util.py [hunk 2]
```diff
@@ -184,21 +185,60 @@ def get_source_type(db_types, qml_geometry):
         if "ST_GeometryCollection" in db_types:
             result = "GeometryCollection"
     if result is None:
-        raise LaymanError(47, data=f'Unknown combination of QGIS geometry "{qml_geometry}" and DB geometry types {db_types}')
+        raise LaymanError(47,
+                          data=f'Unknown combination of QGIS geometry "{qml_geometry}" and DB geometry types '
```
Comment (index-git) [inline src/layman/layer/qgis/util.py:189]: of QGIS geometry => of **QML** geometry

File: src/layman/layer/qgis/util.py [hunk 3]
```diff
@@ -184,21 +185,60 @@ def get_source_type(db_types, qml_geometry):
...
-        ("/qgis/attributetableconfig/columns/column[@type='field']", "name"),
-        ("/qgis/editable/field", "name"),
-        ("/qgis/labelOnTop/field", "name"),
-    ]:
+    for el_path, attr_name in FIELD_XML_ATTRIBUTES:
         for el in qml.xpath(f'{el_path}[@{attr_name}]'):
             el.attrib[attr_name] = db_commmon.launder_attribute_name(el.attrib[attr_name])
+
+
+def get_attribute_names(qml):
```
Comment (index-git) [inline src/layman/layer/qgis/util.py:213]: get_attribute_names => get_attribute_names_from_qml

File: src/layman/layer/qgis/util.py [hunk 4]
```diff
@@ -184,21 +185,60 @@ def get_source_type(db_types, qml_geometry):
...
+    result = set()
+    for el_path, attr_name in FIELD_XML_ATTRIBUTES:
+        for el in qml.xpath(f'{el_path}[@{attr_name}]'):
+            result.add(el.attrib[attr_name])
+    return result
+
+
+def ensure_attributes(qml, attrs_to_ensure):
```
Comment (index-git) [inline src/layman/layer/qgis/util.py:221]: ensure_attributes => ensure_attributes_in_qml

**Review by index-git — CHANGES_REQUESTED** [review]

File: src/layman/gs_wfs_proxy_test.py [hunk 5]
```diff
@@ -171,33 +175,41 @@ def test_missing_attribute():
...
         with app.app_context():
-            wfs_url = f"http://{settings.LAYMAN_SERVER_NAME}/geoserver/{username}/wfs"
+            wfs_url = f"http://{settings.LAYMAN_SERVER_NAME}/geoserver/{workspace}/wfs"
             old_db_attributes = {}
             old_wfs_properties = {}
-            for layername, attr_names in attr_names_list:
+            for layer, attr_names in attr_names_list:
                 # test that all attr_names are not yet presented in DB table
-                old_db_attributes[layername] = db.get_all_column_names(username, layername)
+                old_db_attributes[layer] = [col.name for col in db.get_all_column_infos(workspace, layer)]
```
Comment (jirik) [inline src/layman/gs_wfs_proxy_test.py:203]: Can be `db.get_all_column_names(username, layername)`

File: src/layman/gs_wfs_proxy_test.py [hunk 6]
```diff
@@ -206,21 +218,27 @@ def wfs_post(username, attr_names_list, data_xml):
 
             new_db_attributes = {}
             new_wfs_properties = {}
-            for layername, attr_names in attr_names_list:
+            for layer, attr_names in attr_names_list:
                 # test that exactly all attr_names were created in DB table
-                new_db_attributes[layername] = db.get_all_column_names(username, layername)
+                new_db_attributes[layer] = [col.name for col in db.get_all_column_infos(workspace, layer)]
```
Comment (jirik) [inline src/layman/gs_wfs_proxy_test.py:223]: Can be `db.get_all_column_names(username, layername)`

**Review by index-git — APPROVED** [review]

**Pull request merged**